### ISSUE TITLE
Incompatibility with Rails 4.2.5.1 from a change to find_templates

### ISSUE THREAD

**Issue by aprescott** [emilioforrer/haml_coffee_assets#146]
I'm creating this issue just in case someone else hits this problem. I know #128 exists and there's no maintainer right now.

Rails 4.2.5.1 was just released, and [it includes a change to `find_templates`](https://github.com/rails/rails/compare/v4.2.5...v4.2.5.1#diff-40f2a5bcf9edfd65a8efe42a05c53736R183). It now takes 5 arguments:

```diff
-    def find_templates(name, prefix, partial, details)
+    def find_templates(name, prefix, partial, details, outside_app_allowed = false)
```

haml_coffee_assets (1.16.0) [uses this method, but defines it with 4 arguments in `resolver.rb`](https://github.com/netzpirat/haml_coffee_assets/blob/a3c2951eca005d4fcce4837b9a97a139fc022c66/lib/haml_coffee_assets/action_view/resolver.rb#L10-L11):

```ruby
class Resolver < ::ActionView::FileSystemResolver
  def find_templates(name, prefix, partial, details)
```

There's thus an arity error.

```
  1) Api::Foo::BarsController#index 
     Failure/Error: respond_with bars
     
     ArgumentError:
       wrong number of arguments (given 5, expected 4)
```

**Comment by marten** [comment]
:+1:

**Comment by aprescott** [comment]
I can't think of a way around this in a repo other than to completely redefine `Resolver#find_templates` to handle the new arg, or potentially even just make it `find_templates(*args)`, since it looks like it doesn't need them.

I'd be happy to open a PR with this fix, although I can't tell if anything is getting merged right now.

**Comment by LPiner** [comment]
Thank you!

**Comment by gordonk** [comment]
Thanks!

**Comment by emilioforrer** [comment]
I published a new version of the gem that should fix this issue

**Comment by P9GIT** [comment]
@emilioforrer

Thanks!!

**Comment by nfm** [comment]
@emilioforrer Thanks :smile: 

Is it possible to merge the commit into master and tag the release? I'm not 100% sure what state things are in currently, but master is still pointing at a3c2951eca005d4fcce4837b9a97a139fc022c66 instead of your commit.

**Comment by masterkain** [comment]
@emilioforrer repo is gone?

**Comment by aprescott** [comment]
> Is it possible to merge the commit into master and tag the release?

I'd appreciate this, too. I have a fairly brittle application that uses haml_coffee_assets and I'd like to be able to see the diff between v1.16.0 and v1.16.1.

**Comment by emilioforrer** [comment]
I only got access to the rubygems, not the github repository, as soon as i can get access to it, i will push the changes

**Comment by JIucToyxuu** [comment]
@emilioforrer see issue https://github.com/netzpirat/haml_coffee_assets/issues/148
Few users already try get access to repo, but without success result.

**Comment by derSascha** [comment]
The fix in 88e02ccb59b53fab881d760d52ba0d265f170d60 breaks all activesupport versions prior 4.2.5.1 (including rails 3.x...). Maybe overriding private methods is not the best idea. A runtime dependency on activesupport would be nice.

**Comment by aprescott** [comment]
@derSascha does it break because of the discussion in https://github.com/netzpirat/haml_coffee_assets/pull/147#discussion-diff-50841750 around `super`?

**Comment by derSascha** [comment]
Yes, its the problem! I can confirm that activesupport 3.2.22.1 and haml_coffee_assets 1.16.1 are incompatible.

**Comment by emilioforrer** [comment]
published the version 1.16.2 that should fix the issues wtih older versions of rails

**Comment by paulmenzel** [comment]
For reference, that search engines can index it, the error with Rails 3.2.22.2 looks like below.

```
ArgumentError: wrong number of arguments (5 for 4)
  File "/home/app/vendor/bundle/ruby/2.1.0/gems/haml_coffee_assets-1.16.0/lib/haml_coffee_assets/action_view/resolver.rb", line 10, in find_templates
  File "/home/app/vendor/bundle/ruby/2.1.0/gems/actionpack-3.2.22.2/lib/action_view/template/resolver.rb", line 46, in block in find_all
[…]
```
